bridges/pangolin-rococo/bridge/src/service/header/rococo_to_pangolin.rs

```diff
@@ -83,5 +149,111 @@ async fn start() -> color_eyre::Result<()> {
             ))
         })?;
 
+    let block_number = last_relayed_rococo_block_in_pangolin.block.header.number;
+    let next_mandatory_block = header_relay
+        .subquery_rococo
+        .next_header(block_number)
+        .await?;
+    let next_para_included_event = header_relay
+        .subquery_parachain_rococo
+        .next_candidate_included_event(block_number, 2105u32)
+        .await?;
+
+    let to_relay = match (next_mandatory_block, next_para_included_event) {
+        (Some(mandatory_block), Some(included_event)) => {
+            let relay_chain_block_hash = header_relay
+                .client_rococo
+                .subxt()
+                .rpc()
+                .block_hash(Some(included_event.included_relay_block.into()))
+                .await?
+                .unwrap()
+                .to_string();
+            if mandatory_block.block_number < included_event.included_relay_block {
+                [
+                    (Some(mandatory_block.block_hash), true),
+                    (Some(relay_chain_block_hash), false),
+                ]
+            } else if mandatory_block.block_number == included_event.included_relay_block {
+                [(Some(mandatory_block.block_hash), true), (None, true)]
+            } else {
+                [
+                    (Some(relay_chain_block_hash), false),
+                    (Some(mandatory_block.block_hash), true),
+                ]
+            }
+        }
+        (None, None) => {
+            tracing::info!(
+                target: "pangolin-rococo",
+                "[header-rococo-to-pangolin] No more header to relay after block: {}",
+                block_number,
+            );
+            return Ok(());
+        }
+        (None, Some(included_event)) => {
+            let relay_chain_block_hash = header_relay
+                .client_rococo
+                .subxt()
+                .rpc()
+                .block_hash(Some(included_event.included_relay_block.into()))
+                .await?
+                .unwrap()
+                .to_string();
+            [(Some(relay_chain_block_hash), false), (None, true)]
+        }
+        (Some(mandatory_block), None) => [(Some(mandatory_block.block_hash), true), (None, true)],
+    };
+
+    for (some_block_hash, is_mandatory) in to_relay.into_iter() {
+        if let Some(block_hash) = some_block_hash {
+            let justification = header_relay
+                .subquery_rococo
+                .find_justification(block_hash.clone(), is_mandatory)
```

Comment:
Is it possible to use the latest justification.

Reply:
I have made some updates that tries to relay mandatories firstly, and that tries to relay non-mandatory headers when there is no next mandatory header.